input/drivers_joypad/udev_joypad.c

```diff
@@ -564,6 +580,24 @@ static bool udev_joypad_init(void *data)
       struct udev_device  *dev = udev_device_new_from_syspath(udev_joypad_fd, name);
       const char      *devnode = udev_device_get_devnode(dev);
 
+      if (devnode != NULL) {
+         sorted[sorted_count].devnode = devnode;
+         sorted[sorted_count].item = item;
+         sorted_count++;
+      } else {
+         udev_device_unref(dev);
+      }
+   }
+
+   // Sort the udev entries by devnode name so that they are created in the proper order
```

Comment:
Just for future reference - always do C comments instead of C++ since we want to be able to compile the code as C89. I will do this manually this time.